File: PHP/Memory.php
<?php

class Memory extends Hardware
{
    private $frequency;
    private $memorySize;
    private $supportsCuda;

    function __construct($frequency = null, $memorySize = null, $supportsCuda = null)
    {
        $this->frequency = $frequency;
        $this->memorySize = $memorySize;
        $this->supportsCuda = $supportsCuda;
    }

    function setFrequency($frequency)
    {
        $this->frequency = $frequency;
    }

    function getFrequency()
    {
        return $this->frequency;
    }

    function setMemorySize($memorySize)
    {
        $this->memorySize = $memorySize;
    }

    function getMemorySize()
    {
        return $this->memorySize;
    }

    function setSupportsCuda($supportsCuda)
    {
        $this->supportsCuda = $supportsCuda;
    }

    function getSupportsCuda()
    {
        return $this->supportsCuda;
    }
}

?>

File: Java/Memory.java
public class Memory extends Hardware {
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    public Memory() {

    }

    public Memory(String frequency, String memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    public String getFrequency() {
        return this.frequency;
    }
    
    public void setMemorySize(String memorySize) {
        this.memorySize = memorySize;
    }
    
    public String getMemorySize() {
        return this.memorySize;
    }
    
    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public String getSupportsCuda() {
        return this.supportsCuda;
    }
}
